storage/innobase/xtrabackup/src/backup_copy.cc

```diff
@@ -1449,7 +1449,31 @@ bool backup_start(Backup_context &context) {
   xb_mysql_query(mysql_connection, "FLUSH NO_WRITE_TO_BINLOG BINARY LOGS",
                  false);
 
-  context.log_status = log_status_get(mysql_connection);
+  lsn_t page_tracking_start_lsn = 0;
+  auto log_status = log_status_get(mysql_connection);
+
+  if (opt_page_tracking &&
+      pagetracking::start(mysql_connection, &page_tracking_start_lsn)) {
+    msg("xtrabackup: pagetracking is started on the server with LSN " LSN_PF
+        "\n",
+        page_tracking_start_lsn);
+    while (true) {
+      if (log_status.lsn_checkpoint >= page_tracking_start_lsn) {
+        msg("xtrabackup: pagetracking: Checkpoint lsn is " LSN_PF
+            " and page_tracking lsn is " LSN_PF "\n",
+            log_status.lsn_checkpoint, page_tracking_start_lsn);
+        break;
+      } else {
```

Comment:
@rahulmalik87 why do we need to wait for page_tracking_lsn to cross the current checkpoint_lsn? a page tracking that is started before the current checkpoint is fine right?

Reply:
It is to ensure background redo log copy thread have copied all redo till page_tracking_lsn. or to ensure the backup stop checkpoint LSN is more than page_tracking_lsn.

Reply:
PXB copies redo logs until log_status.lsn_checkpoint. so page-tracking lsn is more than log_status.lsn_checkpoint we are safe since all redo logs will be copied

Reply:
I think by this time, all InnoDB-related copying is done. ie. redo is copying is done and as well as InnoDB data files.

Reply:
at time of log_status.lsn_checkpoint redo copy is still running in parallel thread. that is only stopped once we have checkpoint from log_status.

Reply:
Hmm, only data copying threads are stopped. I see that redo manager is stopped after this. So please add a proper comment before while loop and close this conversation.